pkg/k8sapi/discovery.go

```diff
@@ -68,15 +68,31 @@ type Controller struct {
 	controller *controller
 	kubeClient kubernetes.Interface
 	myNodeName string
-	synced     bool
 }
 
 // NewController creates a new DiscoveryController
```

Comment:
Please update the docstring comment here to indicate that the function constructs the caches (and connects to the k8s API server to do so) before returning a fully-constructed controller object.